common/src/main/cpp/extensions/openxr_fb_body_tracking_extension_wrapper.cpp

```diff
@@ -0,0 +1,359 @@
+	system_body_tracking_properties.type = XR_TYPE_SYSTEM_BODY_TRACKING_PROPERTIES_FB;
+	system_body_tracking_properties.next = next_pointer;
+	system_body_tracking_properties.supportsBodyTracking = false;
+	return reinterpret_cast<uint64_t>(&system_body_tracking_properties);
+}
+
+godot::Dictionary OpenXRFbBodyTrackingExtensionWrapper::_get_requested_extensions() {
+	godot::Dictionary result;
+	for (auto ext : request_extensions) {
+		godot::String key = ext.first;
+		uint64_t value = reinterpret_cast<uint64_t>(ext.second);
+		result[key] = (godot::Variant)value;
+	}
+	return result;
+}
+
+void OpenXRFbBodyTrackingExtensionWrapper::_on_instance_created(uint64_t instance) {
+	if (fb_body_tracking_ext) {
+		bool result = initialize_fb_body_tracking_extension((XrInstance)instance);
+		if (!result) {
+			UtilityFunctions::print("Failed to initialize fb_body_tracking extension");
+			fb_body_tracking_ext = false;
+		}
+	}
+}
+
+void OpenXRFbBodyTrackingExtensionWrapper::_on_instance_destroyed() {
+	cleanup();
+}
+
+void OpenXRFbBodyTrackingExtensionWrapper::_on_session_created(uint64_t instance) {
+	// Skip if not enabled
+	if (!is_enabled()) {
+		return;
+	}
+
+	// Create the body-tracker handle
+	XrBodyTrackerCreateInfoFB createInfo = { XR_TYPE_BODY_TRACKER_CREATE_INFO_FB };
+	createInfo.bodyJointSet = XR_BODY_JOINT_SET_DEFAULT_FB;
+	XrResult result = xrCreateBodyTrackerFB(SESSION, &createInfo, &body_tracker);
+	if (XR_FAILED(result)) {
+		UtilityFunctions::print("Failed to create body-tracker handle: ", result);
+		return;
+	}
+
+	// Construct the XRBodyTracker if necessary
+	if (xr_body_tracker.is_null()) {
+		xr_body_tracker.instantiate();
+		xr_body_tracker->set_body_flags(XRBodyTracker::BODY_FLAG_UPPER_BODY_SUPPORTED | XRBodyTracker::BODY_FLAG_HANDS_SUPPORTED);
+	}
+}
+
+void OpenXRFbBodyTrackingExtensionWrapper::_on_session_destroyed() {
+	// Skip if no body-tracker handle
+	if (!body_tracker) {
+		return;
+	}
+
+	// Destroy the body-tracker handle
+	XrResult result = xrDestroyBodyTrackerFB(body_tracker);
+	if (XR_FAILED(result)) {
+		UtilityFunctions::print("Failed to destroy body-tracker handle: ", result);
+	}
+	body_tracker = XR_NULL_HANDLE;
+
+	// Unregister the body tracker.
+	if (xr_body_tracker_registered) {
+		XRServer *xr_server = XRServer::get_singleton();
+		if (xr_server) {
+			xr_server->remove_body_tracker("/user/body");
+		}
+	}
+	xr_body_tracker_registered = false;
+}
+
+void OpenXRFbBodyTrackingExtensionWrapper::_on_process() {
+	// Skip if not enabled, or no body-tracker handle
+	if (!is_enabled() || !body_tracker) {
+		return;
+	}
+
+	// Get the next frame time
+	const XrTime next_frame_time = get_openxr_api()->get_next_frame_time();
+	if (next_frame_time == 0) {
+		return;
+	}
+
+	// Construct the expression info struct.
+	XrBodyJointsLocateInfoFB locate_info = { XR_TYPE_BODY_JOINTS_LOCATE_INFO_FB };
+	locate_info.time = next_frame_time;
+	locate_info.baseSpace = (XrSpace)get_openxr_api()->get_play_space();
```

Comment:
We're standardizing on using braced-initialization for enums initialization. See https://github.com/GodotVR/godot_openxr_vendors/pull/100#issuecomment-1981726339 for reference.

Reply:
Switched to brace initialization.